src/vs/editor/contrib/codeAction/browser/types.ts

```diff
@@ -123,6 +136,7 @@ export interface CodeActionTrigger {
 		readonly position: Position;
 	};
 	readonly preview?: boolean;
+	readonly triggerAction?: CodeMenuOpenedFrom;
```

Comment:
I think you should make this required. I just tested this and while it does cause a lot of compile errors, it also reveals cases that should have a `triggerAction` passed in